Extended k8sServiceHost=auto options

tchellomello opened the pull request:
This commit extends the `k8sServiceHost=auto` usage to allow the user to determine the namespace and configmap to be used to retrieve the KubeAPI and KubeAPI port respectively.

This already great for kubeadm-based clusters. Therefore, exposing the configuration would allow other user cases.

The default behavior was not modified when using `k8sServiceHost=auto`.

The new introduced parameters are optional.
```release-note
helm: Allow user to configure the namespace used to look up the kube-api server address and port.
```

qmonnet commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -37,13 +37,23 @@ imagePullSecrets: []
+# @schema
+# type: [null, string]
+# @schema
+# -- (string) When `k8sServiceHost=auto`, allows to customize the namespace that containts `k8sServiceLookupConfigMapName`. It defaults to 'kube-public`.
```
```suggestion
# -- (string) When `k8sServiceHost=auto`, allows to customize the namespace that contains `k8sServiceLookupConfigMapName`. It defaults to 'kube-public`.
```

Spell checker caught the typo here, can you fix it please?

tchellomello committed:
Extended k8sServiceHost=auto options

This commit extends the `k8sServiceHost=auto` usage in which allows the
user to determine the namespace and configmap to be used to retrieve the
KubeAPI and KubeAPI port respectively.

This already great for kubeadm-based clusters. Therefore, exposing the
configuration would allow other user cases.

The default behavior was not modified when using `k8sServiceHost=auto`.

The new introduced parameters are optional.

Co-authored-by: Quentin Monnet <qmo@qmon.net>
Signed-off-by: Marcelo Moreira de Mello <tchello.mello@gmail.com>

pippolo84 reviewed (approved):
Thanks!

qmonnet commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -37,13 +37,23 @@ imagePullSecrets: []
+# @schema
+# type: [null, string]
+# @schema
+# -- (string) When `k8sServiceHost=auto`, allows to customize the namespace that contains `k8sServiceLookupConfigMapName`. It defaults to 'kube-public`.
```
Thanks! I just realise you have a mix of straight and back quotes, could you please fix it?


```suggestion
# -- (string) When `k8sServiceHost=auto`, allows to customize the configMap name. It defaults to `cluster-info`.
k8sServiceLookupConfigMapName: ""
# @schema
# type: [null, string]
# @schema
# -- (string) When `k8sServiceHost=auto`, allows to customize the namespace that contains `k8sServiceLookupConfigMapName`. It defaults to `kube-public`.
```

qmonnet commented on Documentation/helm-values.rst:
```diff
@@ -2357,7 +2357,15 @@
+     - string
+     - ``""``
+   * - :spelling:ignore:`k8sServiceLookupNamespace`
+     - When ``k8sServiceHost=auto``\ , allows to customize the namespace that contains ``k8sServiceLookupConfigMapName``. It defaults to 'kube-public`.
```
These are not up-to-date, you need a new `make update-helm-values`

tchellomello replied:
Yes, I was running the tests locally and also caught this. PR updated! thanks @qmonnet !

qmonnet reviewed (approved):
Thanks for your work!
